jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/canonical/TestCanonicalMetaModel.java

```diff
@@ -0,0 +1,95 @@
+package org.eclipse.persistence.jpa.test.canonical;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.eclipse.persistence.config.SessionCustomizer;
+import org.eclipse.persistence.internal.jpa.deployment.SEPersistenceUnitInfo;
+import org.eclipse.persistence.internal.sessions.AbstractSession;
+import org.eclipse.persistence.jpa.PersistenceProvider;
+import org.eclipse.persistence.sessions.Session;
+import org.junit.Test;
+
+public class TestCanonicalMetaModel {
+
+    @Test
+    public void testUseCanonicalModelFieldName() {
+
+        PersistenceProvider persistenceProvider = new PersistenceProvider();
+        SEPersistenceUnitInfo persistenceUnitInfo = new SEPersistenceUnitInfo();
+
+        // For ASM to do its work creating metaclasses
+        persistenceUnitInfo.setClassLoader(Thread.currentThread().getContextClassLoader());
+
+        // We only programmatically add 'leaf' domain classes to pui
+        persistenceUnitInfo.setManagedClassNames(Arrays.asList(DomainClass.class.getName()));
+        persistenceUnitInfo.setPersistenceUnitName("canonical-test");
+        persistenceUnitInfo.setPersistenceUnitRootUrl(DomainClass.class.getProtectionDomain()
+                .getCodeSource().getLocation());
+
+        Properties properties = new Properties();
+        properties.setProperty("javax.persistence.jdbc.driver", "com.mysql.jdbc.Driver");
+        properties.setProperty("javax.persistence.jdbc.url", "jdbc:mysql://localhost/user");
+        properties.setProperty("javax.persistence.jdbc.user", "user");
+        properties.setProperty("javax.persistence.jdbc.password", "password");
```

Comment:
you want to read these from the system, see https://github.com/eclipse-ee4j/eclipselink/blob/master/jpa/eclipselink.jpa.test.jse/antbuild.xml#L110-L113